app/src/Application/DataProvider.cpp

```diff
@@ -0,0 +1,24 @@
+#include "DataProvider.h"
+
+using namespace Status::Application;
+
+namespace StatusGo = Status::StatusGo;
+
+DataProvider::DataProvider()
+    : QObject(nullptr)
+{
+}
+
+StatusGo::Settings::SettingsDto DataProvider::getSettings() const
+{
+    try {
+        return StatusGo::Settings::getSettings();
```

Comment:
@stefandunca the thing is that setting as well as some other info we need/will need must be accessible on the app level. For this particular case since we don't have user profile class/object introduced yet, also since we don't have other way to get the public key of logged in user (there is now getLoggedInUserData end point for status-go) I used settings data which contains the public key of logged in user.

Somehow we run a bit early into implementation without the foundation layer prepared enough.